tools/tsp-client/src/index.ts

```diff
@@ -0,0 +1,281 @@
+import * as path from "node:path";
+
+import { installDependencies } from "./npm.js";
+import { createTempDirectory, removeDirectory,readTspLocation, getEmitterFromRepoConfig } from "./fs.js";
+import { Logger, printBanner, enableDebug, printVersion } from "./log.js";
+import { compileTsp, discoverMainFile, getEmitterOptions, resolveTspConfigUrl } from "./typespec.js";
+import { getOptions } from "./options.js";
+import { mkdir, writeFile, cp, readFile } from "node:fs/promises";
+import { addSpecFiles, checkoutCommit, cloneRepo, getRepoRoot, sparseCheckout } from "./git.js";
+import { fetch } from "./network.js";
+import { parse as parseYaml } from "yaml";
+
+
+async function sdkInit(
+  {
+    config,
+    outputDir,
+    emitter,
+    commit,
+    repo,
+    isUrl,
+  }: {
+    config: string;
+    outputDir: string;
+    emitter: string;
+    commit: string | undefined;
+    repo: string | undefined;
+    isUrl: boolean;
+  }): Promise<string> {
+  if (isUrl) {
+    // URL scenario
+    const resolvedConfigUrl = await resolveTspConfigUrl(config);
+    Logger.debug(`Resolved config url: ${resolvedConfigUrl.resolvedUrl}`)
+    const tspConfig = await fetch(resolvedConfigUrl.resolvedUrl);
+    const configYaml = parseYaml(tspConfig);
+    if (configYaml["parameters"] && configYaml["parameters"]["service-dir"]){
+      const serviceDir = configYaml["parameters"]["service-dir"]["default"];
+      if (serviceDir === undefined) {
+        Logger.error(`Parameter service-dir is not defined correctly in tspconfig.yaml. Please refer to https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml for the right schema.`)
+      }
+      Logger.debug(`Service directory: ${serviceDir}`)
+      let additionalDirs: string[] = [];
+      if (configYaml["parameters"]["dependencies"] && configYaml["parameters"]["dependencies"]["additionalDirectories"]) {
+        additionalDirs = configYaml["parameters"]["dependencies"]["additionalDirectories"];
+      }
```

Comment:
the nullish coalescing operator is your friend here
```suggestion
      const additionalDirs: string[] = configYaml?.parameters?.dependencies?.additionalDirectories ?? [];
```